src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -531,11 +576,10 @@ private ClassDesc(String qualifiedName, int depth, DetailAST classAst) {
             final DetailAST modifiers = classAst.findFirstToken(TokenTypes.MODIFIERS);
             declaredAsFinal = modifiers.findFirstToken(TokenTypes.FINAL) != null;
             declaredAsAbstract = modifiers.findFirstToken(TokenTypes.ABSTRACT) != null;
-        }
-
-        /** Adds private ctor. */
-        private void registerPrivateCtor() {
-            withPrivateCtor = true;
+            withImplicitConstructor =
+                    classAst.getLastChild().findFirstToken(TokenTypes.CTOR_DEF) == null;
+            privateClass = classAst.findFirstToken(TokenTypes.MODIFIERS)
```

Comment:
`classAst.findFirstToken(TokenTypes.MODIFIERS)` => `modifiers`

Also move this closes to the other modifier booleans and follow their pattern as `declaredAsPrivate`

Reply:
Done

Reply:
@rnveach ping

Reply:
> Also move this closes to the other modifier booleans and follow their pattern as declaredAsPrivate

Not done.

Reply:
done